### DISCUSSION TITLE
Why does Hud Overlay affect performance?

### DISCUSSION THREAD

**Discussion by NotGumballer91** [ppy/osu#15937] [Q&A] [answered]
I've tried playing without hud overlay and my input felt better and responsive and checked whether it was placebo or not.
Turns out it does improve it significantly.

https://user-images.githubusercontent.com/64581009/144719331-4ca2fad0-d483-4be4-9dfb-16f7505b3175.mp4

I'm just unsure why this affects performance more than stable. Is this intended/expected?

**Comment by bdach** [comment, marked as answer]
you can press Ctrl+F2 to display a performance overlay with a breakdown of the 5 components that consume the most frame time (should be somewhere near the bottom).

as for your question, performance is a constant work in progress. gameplay in particular will receive optimisation passes going forward, as some of the more recent additions may not have been implemented in the most optimised way possible (things like PP/UR counter, for one instance). the goal is zero allocations in gameplay, as well as frame times as low as they can feasibly go.

**Comment by peppy** [comment]
Hard to make any calls without knowing what hardware you're running on. If you're playing at above 60hz, then the game is not able to render fast enough for your display device currently. This could either be due to an issue our end, or underpowered hardware. Knowing your system specifications (especially GPU) would be helpful.
